x/gov/keeper/deposit.go

```diff
@@ -162,6 +162,47 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 	return activatedVotingPeriod, nil
 }
 
+// BurnAndSendDepositsToCommunityPool will burn the (deposits * proposal_cancel_burn_rate) amount of proposal deposits
+// and send remaining deposits of the proposal to community pool.
+func (keeper Keeper) BurnAndSendDepositsToCommunityPool(ctx sdk.Context, proposalID uint64, totalDeposits []sdk.Coin) error {
+	store := ctx.KVStore(keeper.storeKey)
+
+	proposalCancelBurnRate := keeper.GetParams(ctx).ProposalCancelBurnRate
+	// burn the deposits * proposal_cancel_burn_rate amount from proposal deposits (gov module)
+	burnRate := sdk.MustNewDecFromStr(proposalCancelBurnRate)
+
+	var burnDepositAmount sdk.Coins
+
+	if burnRate.IsPositive() {
+		for _, deposit := range totalDeposits {
+			burnAmount := sdk.NewCoin(
+				deposit.Denom,
+				sdk.NewDecFromInt(deposit.Amount).Mul(burnRate).RoundInt(),
+			)
+			burnDepositAmount = burnDepositAmount.Add(burnAmount)
+		}
+	}
+
+	// burn the deposits
+	if !burnDepositAmount.IsZero() {
+		err := keeper.bankKeeper.BurnCoins(ctx, types.ModuleName, burnDepositAmount)
+		if err != nil {
+			return err
+		}
+	}
+
+	// send (deposits - burnAmount) to community pool from proposal deposits (gov module)
+	communityPoolAmount := sdk.NewCoins(totalDeposits...).Sub(burnDepositAmount...)
```

Comment:
```suggestion
	communityPoolAmount := sdk.Coins(totalDeposits).Sub(burnDepositAmount...)
```

Reply:
But please double check if we can do it... might be better to update the function argument to assure that coins are sorted.